### PR TITLE
Updates to Combobox: PD-1991 (PD-2024, PD-2011, PD 1910, PD-1907, PD-1860)

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Various bug fixes & improvements to Combobox.

🎟 _Jira tickets:_ 
[Combobox fixes Epic: PD-1991](https://jira.mongodb.org/browse/PD-1991)

 - [x] [PD-2024](https://jira.mongodb.org/browse/PD-2024): [DOCS] Improves wording of `initialValue` prop description
 - [x] [PD-2011](https://jira.mongodb.org/browse/PD-2011): [DOCS] Remove incorrect Guidelines, display Combing Soon
 - [x] [PD-1910](https://jira.mongodb.org/browse/PD-1910): [BUG] Fix padding for `overflow=scroll-x`
 - [x] [PD-1907](https://jira.mongodb.org/browse/PD-1907): [BUG] Fix wrapJSX function
 - [x] [PD-1860](https://jira.mongodb.org/browse/PD-1860): [BUG] Reduce min-width

## These will be in a separate PR(s)
 - [ ] [~PD-1737~](https://jira.mongodb.org/browse/PD-1739): ~[FEAT] Preserve focus when deleting a chip~
 - [ ] [~PD-1801~](https://jira.mongodb.org/browse/PD-1801): ~[FEAT] Add functionality to disable a single option~
 - [ ] [~PD-1771~](https://jira.mongodb.org/browse/PD-1771): ~[FEAT] Add support for Size variants~
 - [ ] [~PD-1739~](https://jira.mongodb.org/browse/PD-1739): ~[FEAT] All options visible when a single-select opens~
 - [ ] [~PD-2046~](https://jira.mongodb.org/browse/PD-2046): ~[FEAT] Preserve native backdrop click behavior~ #1256 

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/combobox/src/Combobox.styles.ts [hunk 1]
```diff
@@ -64,6 +69,11 @@ export const comboboxParentStyle = ({
         var(--lg-combobox-padding-x) var(--lg-combobox-padding-y)
         ${overflow === 'scroll-x' ? '0' : 'var(--lg-combobox-padding-x)'};
       width: var(--lg-combobox-width);
+      // TODO: Clean this up 🤮
```
Comment (TheSonOfThomp) [inline packages/combobox/src/Combobox.styles.ts:72]: I'll clean this up when I do the refresh

File: packages/combobox/README.md [hunk 2]
```diff
@@ -81,7 +81,7 @@ import { Combobox, ComboboxOption } from '@leafygreen-ui/combobox';
...
-| `initialValue`           | `Array<string>`, `string`                     | The initial selection. Must be a string for a single-select, or an array of strings for multiselect. Changing the `initialValue` after initial render will not change the selection.                                                                                                                                                                                                                                                                                |                          |
+| `initialValue`           | `Array<string>`, `string`                     | The `value` of the initial selection. Mapped to the `value` prop of the provided options. (See note on default `value`s) Must be a string for a single-select, or an array of strings for multiselect. Changing the `initialValue` after initial render will not change the selection.                                                                                                                                                                              |                          |
```
Comment (shaneeza) [inline packages/combobox/README.md:84]: If I understand this, what you're saying is that `initialValue` and `value` should follow the same format? If thats the case, it's not that clear to me. Maybe you can be more explicit and say they should follow the same format?

Comment (TheSonOfThomp) [reply packages/combobox/README.md]: `initialValue` and/or `value` should match the `value` prop of a `ComboboxOption`

Comment (shaneeza) [reply packages/combobox/README.md]: I see, that makes it clearer

File: packages/combobox/README.md [hunk 3]
```diff
@@ -81,8 +81,8 @@ import { Combobox, ComboboxOption } from '@leafygreen-ui/combobox';
...
-| `value`                  | `Array<string>`, `string`                     | The controlled value of the Combobox. Must be a string for a single-select, or an array of strings for multiselect. Changing `value` after initial render _will_ affect the selection.                                                                                                                                                                                                                                                                              |                          |
+| `initialValue`           | `Array<string>`, `string`                     | The initial selection. Must be a string (or array of strings) that matches the `value` prop of a `ComboboxOption`. Changing the `initialValue` after initial render will not change the selection.                                                                                                                                                                                                                                                                  |                          |
```
Comment (shaneeza) [inline packages/combobox/README.md:84]: 👍🏽

**Review by shaneeza — APPROVED** [review]

**Pull request merged**